src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2767,13 +2793,7 @@ private unsafe void SendManifest(byte[]? rawManifest)
             if (!member.Module.Assembly.ReflectionOnly && (flags & EventManifestOptions.AllowEventSourceOverride) == 0)
             {
                 // Let the runtime to the work for us, since we can execute code in this context.
-                Attribute? firstAttribute = null;
-                foreach (object attribute in member.GetCustomAttributes(attributeType, false))
-                {
-                    firstAttribute = (Attribute)attribute;
-                    break;
-                }
-                return firstAttribute;
+                return member.GetCustomAttribute(attributeType, inherit: false);
```

Comment:
Note the potential for behavior change here . Previously the first attribute was returned if there were multiple. But now it appears to throw:

https://github.com/dotnet/runtime/blob/8636dc8a76d687169db64033b71ec9f79d9ba4b5/src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs#L505-L515

Looking at the callers of `GetCustomAttributeHelper`, it appears it is only called with `AllowMultiple=false` attributes. But just wanted to call this out.

Reply:
Added a `Debug.Assert` on type to confirm this contract in code